nethttp/server.go

```diff
@@ -22,6 +22,7 @@ func (w *statusCodeTracker) WriteHeader(status int) {
 type mwOptions struct {
 	opNameFunc    func(r *http.Request) string
 	componentName string
+	spanObserver  func(span opentracing.Span, r *http.Request)
```

Comment:
MWOption can only used to set operationName or ComponentName, so i add the spanObserver to run the func.